include/envoy/http/async_client.h

```diff
@@ -50,18 +83,64 @@ class AsyncClient {
     virtual void cancel() PURE;
   };
 
+  /**
+   * An in-flight HTTP stream.
+   */
+  class Stream {
+  public:
+    virtual ~Stream() {}
+
+    /***
+     * Send headers to the stream.
+     * @param headers supplies the headers to send.
+     * @param end_stream supplies whether this is a header only request.
+     */
+    virtual void sendHeaders(HeaderMap& headers, bool end_stream) PURE;
+
+    /***
+     * Send data to the stream.
+     * @param data supplies the data to send.
+     * @param end_stream supplies whether this is the last data.
+     */
+    virtual void sendData(Buffer::Instance& data, bool end_stream) PURE;
```

Comment:
It's programming error. Looks like an error code should be returned.
Without changing the method signature, it should be ignored with an error log?